src/main/java/ru/skypro/homework/controller/AdsController.java

```diff
@@ -14,158 +17,146 @@
+            @ApiResponse(responseCode = "401", description = "Unauthorized"),
+            @ApiResponse(responseCode = "404", description = "Not Found")
+    })
     public ResponseEntity<ExtendedAdDTO> getAdById(@PathVariable int adId) {
         return ResponseEntity.ok().body(adsService.getAdById(adId));
     }
 
     // удаление объявления по id
-    @Operation(
-            operationId = "removeAdById",
-            summary = "Удалить объявление по id",
-            tags = {"Объявления"},
-            responses = {
-                    @ApiResponse(responseCode = "204", description = "No Content"),
-                    @ApiResponse(responseCode = "401", description = "Unauthorized"),
-                    @ApiResponse(responseCode = "403", description = "Forbidden"),
-                    @ApiResponse(responseCode = "404", description = "Not Found")
-            }
-    )
-    @PreAuthorize("hasRole('ADMIN') or (hasRole('USER') and" +
-            "@adsService.getAdById(#adId).email == authentication.principal.username)")
+
+    @Operation(summary = "Удалить объявление по id")
 
     @DeleteMapping("/{adId}")
-    public void removeAdById(@PathVariable int adId) {
-        adsService.deleteAdsById(adId);
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "204", description = "No Content"),
+            @ApiResponse(responseCode = "401", description = "Unauthorized"),
+            @ApiResponse(responseCode = "403", description = "Forbidden"),
+            @ApiResponse(responseCode = "404", description = "Not Found")
+    })
+    @PreAuthorize("hasRole('ADMIN') or " +
+            "@adsService.getAdById(#adId).email == authentication.name")
+
+    public ResponseEntity<?> removeAdById(@PathVariable int adId, Authentication authentication) throws IOException {
+        if (adsService.checkAccessToAd(adId, authentication.getName())) {
+            adsService.deleteAdsById(adId);
+            return new ResponseEntity<>(HttpStatus.OK);
+        }
+        log.info("Отказано в доступе для " + authentication.getName());
+        return new ResponseEntity<>(HttpStatus.FORBIDDEN);
     }
 
-    @Operation(
-            operationId = "updateAd",
-            summary = "Обновить объявление по id",
-            tags = {"Объявления"},
-            responses = {
-                    @ApiResponse(responseCode = "200", description = "OK"),
-                    @ApiResponse(responseCode = "401", description = "Unauthorized"),
-                    @ApiResponse(responseCode = "403", description = "Forbidden"),
-                    @ApiResponse(responseCode = "404", description = "Not Found")
-            }
-    )
-    @PreAuthorize("hasRole('ADMIN') or (hasRole('USER') and" +
-            "@adsService.getAdById(#adId).email == authentication.principal.username)")
-
+    @Operation(summary = "Обновить объявление по id")
+    @PreAuthorize("hasRole('ADMIN') or " +
+            "@adsService.getAdById(#adId).email == authentication.name")
```

Comment:
```suggestion
            "@adsService.getAdById(#adId).username == authentication.name")
```
потому что поле в сущности User называется username, а не email